apputils/config_validator.py

```diff
@@ -0,0 +1,52 @@
+"""Validator for config files."""
+
+
+class ConfigValidator(object):
+    sections_required = ["task_config", "log_config", "schedule",
+                         "env_literal", "env_expand"]
+    task_config_required = ['task_name', 'image', 'retry', 'read_only']
+    log_config_required = ['task_started', 'task_finished',
+                           'task_retried', 'task_failed']
+    schedule_required = ['minute', 'hour', 'day_of_week',
+                         'day_of_month', 'month_of_year']
+
+    @classmethod
+    def validate_config(cls, config):
```

Comment:
pull this if statement into a function so it can be separately tested when there are missing services, etc. ie. if self.missing_services? return

separate the rest of missing sections logic into functions that can be individually tested